### PR TITLE
DOC: Extended the documentation for `DataFrame.sort_values()`

### PR DESCRIPTION
**Pull request by aallahyar**
The current docstring for [`DataFrame.sort_values()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sort_values.html) is lacking clarity, and also contains links that are not converted to hyperlinks. I extended and rephrased part of the documentation for this method. In particular:

- Added further explanation to compare the single-column vs. multi-column sorting,
- Extended explanation for customized sorting via `key` argument,
- Provided a simplified version of `natsort` example, with added explanation.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -6891,6 +6913,8 @@ def sort_values(
         3   48hr     40
         4   96hr     50
         >>> from natsort import index_natsorted
+        >>> index_natsorted(df["time"])
+        [0, 3, 2, 4, 1]
         >>> df.sort_values(
         ...     by="time", key=lambda x: np.argsort(index_natsorted(df["time"]))
```
Comment (datapythonista) [inline pandas/core/frame.py:6919]: This example seems wrong. The `df["time"]` inside of the lambda should be the `x` received as the parameter. Do you mind updating it since you are improving this docstring?

**Review by datapythonista — COMMENTED** [review]
Thanks for improving this docstring @aallahyar. Your changes look great, but while reviewing I spotted an error in an example in the original docstring. If you want to fix it, that would be amazing.

Also, for next time, it's in general better to use a branch to open PRs. If you only contribute this PR it shouldn't make a difference, but if you open a second PR to pandas, things will become wild with git (changes here will be mixed with the new PR...).

**Comment by datapythonista** [comment]
The rendered version of the changes here is available [here](https://pandas.pydata.org/preview/pandas-dev/pandas/57678/docs/reference/api/pandas.DataFrame.sort_values.html), in case it's useful.

**Comment by aallahyar** [comment]
Thanks @datapythonista for reviewing the update!

Would be happy to fix the issue. Please take a look.

Also thank you for the guidance!
It is indeed my first contribution to a large project, so I have a lot to learn 😄

File: pandas/core/frame.py [hunk 2]
```diff
@@ -6891,15 +6913,39 @@ def sort_values(
...
+        ...     key=natsort_keygen(),
+        ... )
+            time  value
+        0    0hr     10
+        3   48hr     40
+        2   72hr     30
+        4   96hr     50
+        1  128hr     20
+
+        .. warning::
```
Comment (datapythonista) [inline pandas/core/frame.py:6943]: I understand why you're writing this, but feels like a bit out of topic in the docstring for pandas' `sort_values`. Personally, I found that the last example may be useful for natsort users, but it doesn't add much to explain the `sort_values` method, which is the point of this PR.

If you don't think natsort is reliable, maybe better to just keep the example with str.lower.

Comment (aallahyar) [reply pandas/core/frame.py:6943]: Thanks @datapythonista, I see your point.
Meanwhile, I do see the value of showing how `natural sorting` can be added to `sort_values`. This example was how I learned about `natsort` in the first place.

By the way, the issue that I am currently warning about is actually originating from `pd.MultiIndex` (and not `natsort` per se) where `pd.MultiIndex` silently (i.e., no warning, no errors) creates a single-level index if the provided tuples do not have the same dimensions (see the linked issue above).

In short, would be happy to follow your suggestion. I will reduce the examples, and provide a small reference to `natsort` for anyone who might need it. Would you agree with this?

Comment (datapythonista) [reply pandas/core/frame.py]: I think showing an example with `key` is good, and if natsort has a function we can already use and you think it's a relevant library for users I'm surely happy to have it. My point was that we had two examples showing pretty much the same pandas functionality, and showing different natsort functionalities, the second requiring a warning because it was buggy or misleading. So, probably better to remove the second example. Users reading our page are likely to be interested in pandas functionality  if they discover natsort that's great, but for more natsort examples they can go to natsort documentation.

In summary, happy to have natsort used, but IMHO better not to show advanced natsort things that aren't relevant to pandas users not interested in natsort, or that require a warning. Does it make sense?

Comment (aallahyar) [reply pandas/core/frame.py]: Of course. Sure, lets go with a single example then.
Please see if you agree with the latest update.

Thanks :)

**Comment by aallahyar** [comment]
@datapythonista could you please let me know if any other modification is needed for this PR?

Thanks :)

**Review by datapythonista — APPROVED** [review]
Thanks a lot for the work on this @aallahyar very nice. Looks perfect now (I'll let a second core dev have a look before merging).

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @aallahyar
